lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -505,6 +507,69 @@ def add_marketplace_data(post, options)
         end
       end
 
+      def add_level_two_three_data(post, options)
+        post[:processing] ||= {}
+        post[:customer] ||= {}
+
+        # American Express only supports Level 2 data.
+        # Only is required add items info in lvl2 data for amex
+        add_items(post, options)
+        add_level_two_data(post, options)
+        add_level_three_data(post, options)
+        add_shipping_data(post, options)
+      end
+
+      def add_items(post, options)
+        items = build_items(options[:line_items])
+        post[:items] = items unless items.nil?
+      end
+
+      def add_level_two_data(post, options)
+        post[:customer][:tax_number] = options[:tax_number] # field no require for amex
+
+        post[:processing].merge!(
+          {
+            order_id: options.dig(:processing, :order_id),
```

Comment:
❓ Question:

Is expected that lvl2 data to haave nested values ?

in this case you are asking for options[:processing][:order_id]

Reply:
In this case is to avoid confusing with the order_id in the transaction, I can change it to a simple value called invoice_id and could work too.